ReactiveCocoa/Swift/Signal.swift

```diff
@@ -14,12 +14,14 @@ import Result
 /// progress,” like notifications, user input, etc. To represent streams that
 /// must first be _started_, see the SignalProducer type.
 ///
-/// Signals do not need to be retained. A Signal will be automatically kept
-/// alive until the event stream has terminated.
+/// A Signal is kept alive until either of the following happens:
+///    1. its input observer receives a terminating event; or
+///    2. it has no active observers, and is not being retained.
 public final class Signal<Value, Error: ErrorProtocol> {
 	public typealias Observer = ReactiveCocoa.Observer<Value, Error>
 
-	private let atomicObservers: Atomic<Bag<Observer>?> = Atomic(Bag())
+	private var generatorDisposable: Disposable?
```

Comment:
`self` is used in constructing the observer passed into the generator, and the setting of this comes after. So probably not until we have delayed initialisation.